src/components/Comments/TextField.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  FC, ChangeEvent, useState, useEffect,
+} from 'react';
+import classNames from 'classnames';
+import { FieldIcon, Field } from '../../types';
+
+type Props = {
+  field: Field;
+  type: 'text' | 'textarea';
+  id: string;
+  label: string;
+  placeholder: string;
+  name: 'name' | 'email' | 'body';
+  icon?: FieldIcon;
+  isErrorIcon?: boolean;
+  errorMessage: string;
+  isSubmitted: boolean;
+  isReseted: boolean;
+  value: string;
+  onChange: (value: string) => void;
+  setIsSubmitted: (value: boolean) => void;
+};
+
+export const TextField: FC<Props> = (props) => {
+  const {
+    field,
+    type,
+    id,
+    label,
+    placeholder,
+    name,
+    icon,
+    isErrorIcon,
+    errorMessage,
+    value,
+    isSubmitted,
+    isReseted,
+    setIsSubmitted,
+    onChange,
+  } = props;
+
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    if (isSubmitted) {
+      setIsError(!value);
+      setIsSubmitted(false);
+    }
+  }, [isSubmitted]);
+
```

Comment:
This useEffect hook might have a missing dependency 'onChange'. Consider adding it to the dependency array to avoid unnecessary re-renders or bugs due to stale data.